internal/command/docker_test.go

```diff
@@ -273,6 +273,109 @@ func TestCmdEnginePodman(t *testing.T) {
 	}
 }
 
+func TestAppendEnv(t *testing.T) {
+	type args struct {
+		args        []string
+		env         map[string]string
+		quoteNeeded bool
+	}
+	tests := []struct {
+		name      string
+		args      args
+		wantStart []string
+		wantEnd   [][2]string
+	}{
+		{
+			name: "empty",
+			args: args{
+				args:        []string{},
+				env:         map[string]string{},
+				quoteNeeded: true,
+			},
+			wantStart: []string{},
+			wantEnd:   [][2]string{},
+		},
+		{
+			name: "quote needed",
+			args: args{
+				args:        []string{},
+				env:         map[string]string{"VAR": "value"},
+				quoteNeeded: true,
+			},
+			wantStart: []string{},
+			wantEnd:   [][2]string{{"-e", "VAR=value"}},
+		},
+		{
+			name: "quote not needed",
+			args: args{
+				args:        []string{},
+				env:         map[string]string{"VAR": "value"},
+				quoteNeeded: false,
+			},
+			wantStart: []string{},
+			wantEnd:   [][2]string{{"-e", "VAR=value"}},
+		},
+		{
+			name: "multiple",
+			args: args{
+				args:        []string{},
+				env:         map[string]string{"VAR": "value", "VAR2": "value2"},
+				quoteNeeded: true,
+			},
+			wantStart: []string{},
+			wantEnd:   [][2]string{{"-e", "VAR=value"}, {"-e", "VAR2=value2"}},
+		},
+		{
+			name: "multiple with args",
+			args: args{
+				args:        []string{"arg1", "arg2"},
+				env:         map[string]string{"VAR": "value", "VAR2": "value2"},
+				quoteNeeded: true,
+			},
+			wantStart: []string{"arg1", "arg2"},
+			wantEnd:   [][2]string{{"-e", "VAR=value"}, {"-e", "VAR2=value2"}},
+		},
+		{
+			name: "multiple with args and equal sign require quoting values",
+			args: args{
+				args:        []string{"arg1", "arg2"},
+				env:         map[string]string{"VAR": "value", "VAR2": "value2=2"},
+				quoteNeeded: true,
+			},
+			wantStart: []string{"arg1", "arg2"},
+			wantEnd:   [][2]string{{"-e", "VAR=value"}, {"-e", "VAR2=\"value2=2\""}},
+		},
+		{
+			name: "multiple with args and equal sign do not require quoting values",
+			args: args{
+				args:        []string{"arg1", "arg2"},
+				env:         map[string]string{"VAR": "value", "VAR2": "value2=2"},
+				quoteNeeded: false,
+			},
+			wantStart: []string{"arg1", "arg2"},
+			wantEnd:   [][2]string{{"-e", "VAR=value"}, {"-e", "VAR2=value2=2"}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := AppendEnv(tt.args.args, tt.args.env, tt.args.quoteNeeded)
+			for i, v := range tt.wantStart {
+				assert.Equal(t, v, got[i])
+			}
+			for i := len(tt.wantStart); i < len(got); i += 2 {
```

Comment:
isn't `len(tt.wantStart)` always going to be the same as `i`?
It took a while to realise this was contiguous, so I would suggest that I spans the two for loops so that is clear.

I also wondered, wouldn't it make sense to simplfy to just have all the output in a single `want` that is iterated?

Reply:
The issue is that you can't rely on hash map order, which is why there is two loops. One for the part that should be stable and one for the part that can vary.

Reply:
Oh I see. But that wouldn't impact the usage of i - a single variable would better indicate that it is checking every item in the slices, even though split over two loops?

Reply:
I am not sure what you exactly mean. I reused the same name (`i`), but did want to use the cleaner, more readable, call to `range` when possible.

Reply:
This actually make the code a bit more complex to read. `range` does exit the loop with i == len(tt.wantStart) - 1, but if there is nothing in tt.wantStart, it exit with i == 0. So before I can use it I have to add a test to see if len(tt.wantStart) != 0, to add 1 back. This is a lot less readable at the end that with the current code I think. Do you have any suggestion on how to solve that? I can only imagine with going with the C style loop syntax instead of `range`.

Reply:
I think you could increment `i` inside the range loop too, if that's better?

Reply:
Do you mean that i would not be managed by range, but manually? Will have to try that.

Reply:
Not a big fan, but if that's what you had in mind, I am fine with it.

Reply:
The notation may not be totally awesome, but it helps to show we are actually only iterating once, despite two loops.
Thanks for the update.